lib/collection/src/collection/shard_transfer.rs

```diff
@@ -78,6 +75,7 @@ impl Collection {
             let initial_state = match shard_transfer.method.unwrap_or_default() {
                 ShardTransferMethod::StreamRecords => ReplicaState::Partial,
                 ShardTransferMethod::Snapshot => ReplicaState::PartialSnapshot,
+                ShardTransferMethod::WalDelta => ReplicaState::Partial,
```

Comment:
The problem is that if we don't use `Partial`, we need to add another replica set state. @generall wanted to prevent this if possible.

Reply:
One interesting property is that we'll always send those "intermixed" operations as part of the WAL diff too.

Reply:
Yeah, sure, it's just pointless. Why can't we reuse `PartialSnapshot` for this, though?

Reply:
Because transition from Dead -> `Anything` is not atomic. There will be a gap, where the updates might be send to only one node, so w either need to explicitly wait in `Patrial` state, or ensure that all nodes have received consensus updates

Reply:
I think we can't use Partial while doing diff transfer. Seems like I figured a case where it may break:Here's WAL on node A:

```
clock: A-1, point id: 1, payload: false
clock: A-2, point id: 2, payload: false
...
clock: A-1000, point id: 1000, payload: false
clock: A-1001, update all matching "payload: false" to "payload: true"
```

We are doing diff transfer from node A to node B starting from clock: 1. Here's current WAL on node B:

```
clock: A-1, point id: 1, payload: false
clock: A-2, point id: 2, payload: false
...
clock: A-500, point id: 500, payload: false
(so, ATM we transferred half of the diff)
```

Now imagine that on node C we send delete all matching "payload: false".

- Node B is in Partial state
- So this operation will be "propagated" to both node A and node B
- All 500 points that are currently in the storage on node B have payload: false
- So all of them would be deleted
- Then we will receive points 501-1000 and operation A-1001 (update "payload: false" to "payload: true") and operation from the node C (delete all "payload: false")
- But by this time half of the points on node B are already deleted

Reply:
> Now imagine that on node C...
Also note, that we don't even need node C for this example. We can do this operation directly on node A, and it will work exactly the same way. (Using separate node C just seemed a bit more "natural".)

Reply:
> In this state we currently reject incoming operations, so we cannot forward operations from a queue proxy. Because of it, we cannot transfer the WAL diff in this state.

Can we refactor so that in `PartialSnapshot` we reject all operations, except if they have `force: true`? Feels like it may solve all of our problems with above scenario.

Reply:
We'll use `PartialSnapshot` for now, and migrate to the `Recovery` state in Qdrant 1.9.
Read more about this here: https://github.com/qdrant/qdrant/pull/3659